ur_robot_driver/src/rtde/rtde_client.cpp

```diff
@@ -47,14 +47,38 @@ RTDEClient::RTDEClient(std::string robot_ip, comm::INotifier& notifier, const st
 
 bool RTDEClient::init()
 {
+  // A running pipeline is needed inside setup
+  pipeline_.init();
   pipeline_.run();
+
+  uint16_t protocol_version = negotiateProtocolVersion();
+  parser_.setProtocolVersion(protocol_version);
+
+  queryURControlVersion();
+  if (urcontrol_version_.major < 5)
+  {
+    max_frequency_ = CB3_MAX_FREQUENCY;
```

Comment:
Could be nice to notify users of this. Or at least add a `ROS_DEBUG..(..)`.

Reply:
It actually is still there. However, it got refactored into the `setOutputs()` function which is called directly afterwards.

Do you think, that would be nice to do, nonetheless? Output would then be something like
```
[ DEBUG] [1572871771.209652246]: Setting RTDE communication frequency to 125 Hz, as using a CB3 robot.
[ INFO] [1572871771.209652246]: Setting up RTDE communication with frequency 500.000000
```